modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/SecondaryStoreUserLoginTestCase.java

```diff
@@ -0,0 +1,269 @@
+                    userStoreConfigUtils.getJDBCUserStoreProperties(USER_STORE_DB_NAME));
+            userStoreConfigAdminServiceClient.addUserStore(userStoreDTO);
+            Thread.sleep(5000);
+        } catch (Exception e) {
+            log.error(e);
+        }
+        boolean isSecondaryUserStoreDeployed = userStoreConfigUtils.waitForUserStoreDeployment(
+                userStoreConfigAdminServiceClient, DOMAIN_ID);
+
+        if (isSecondaryUserStoreDeployed) {
+
+//          Creating users in the primary and secondary user stores
+            try {
+                addUserIntoJDBCUserStore(primUsername, primPassword, false);
+                addUserIntoJDBCUserStore(secUsername, secPassword, true);
+            } catch (Exception e) {
+                log.error(e);
+            }
+
+//          Creating and starting application on tomcat
+            initAndCreatePlaygroundApplication();
+            startTomcat();
+            client = HttpClientBuilder.create().setDefaultCookieStore(cookieStore).build();
+
+//          To generate sessionDataKey
+            try {
+                testSendAuthenticationRequest(playgroundApp, client);
+            } catch (Exception e) {
+                log.error(e);
+            }
+        } else {
+            log.error("Secondary user store is not deployed");
+        }
+    }
+
+    @DataProvider(name = "userCredentialProvider")
+    public static Object[][] userCredentialProvider() {
+
+        return new Object[][]{
+                {primUsername, primPassword},
+                {secUsername, secPassword},
+        };
+    }
+
+    @Test(groups = "wso2.is", description = "Check the secondary user store user login flow",
+            dataProvider = "userCredentialProvider")
+    public void testUserLogin(String username, String password) {
+
+        try {
+            testAuthentication(playgroundApp, username, password);
+        } catch (Exception e) {
+            log.error("Error: " + e);
+        }
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void atEnd() throws Exception {
+
+        stopTomcat();
+        userStoreConfigAdminServiceClient.deleteUserStore(DOMAIN_ID);
+        userMgtClient.deleteRole(primaryUserRole);
+        userMgtClient.deleteUser(primUsername);
+        deleteApplication(playgroundApp);
+        clear();
+    }
+
+    private void testAuthentication(OIDCApplication application, String username, String password) throws IOException {
+
+        HttpResponse response = sendLoginPostForCustomUsers(client, sessionDataKey, username, password);
+        Assert.assertNotNull(response, "Login request failed for " + application.getApplicationName()
+                + ". response " + "is null.");
+        Header locationHeader = response.getFirstHeader(OAuth2Constant.HTTP_RESPONSE_HEADER_LOCATION);
+        Assert.assertNotNull(locationHeader, "Login response header is null for "
```

Comment:
What do you mean "Login response header"? You are checking for location header.

Reply:
Just checking location header should not guarantee that the login is completed successfully.

Reply:
Now checking using the authorization code